core/stdlib/std.ncl

```diff
@@ -792,6 +792,55 @@
   },
 
   contract = {
+    Assert
+      | doc m%"
+        A contract that checks if its argument is the boolean true.
+
+        We could get by just using arrays of boolean tests and `std.array.all`.
+
+        However, this contract provides fine-grained error reporting,
+        pinpointoing the exact expression that failed in an array of tests, as
+        opposed to the pure boolean solution.
+
+        # Examples
+
+        ```nickel
+        1 == 2 | std.contract.Assert
+         => error: contract broken by a value
+
+        1 == 1 | std.contract.Assert
+         => true
+        ```
+        "%
+        = fun l x => x || %blame% l,
+
+    # We can't use a static type because x | Assert is not of type bool. We
+    # could use additional contracts to make it work but it's not worth the
+    # hassle.
+    check
```

Comment:
I think `check` is a bit too generic for the stdlib. When using it internally for Nickel tests it's fine, but if it's to move to the stdlib, I would pick a name that is more explicit. Maybe `assert_all` is more descriptive, for example. Same for the description, I would make it less "test"-oriented, and just says that this checks that the conditions provided as an array of boolean are true, or it will fail with a contract error pointing to the one that failed.

Reply:
yea I agree, assert_all makes more sense. I don't see a reason not to add it to the stdlib, since it's such a common pattern